src/entities/transaction-log.entities.ts

```diff
@@ -0,0 +1,20 @@
+import { IBaseEntity } from './base.entity';
+import { IProfile } from './profile.entities';
+
+export interface TransactionLogs<D extends IBaseEntity> extends IBaseEntity {
+  key?: string;
+  operator: string | IProfile;
+  date: string;
+  kind: 'operation' | 'post-operation' | 'query' | 'console';
```

Comment:
types of logs:

1.  operation - takes argument as payload for logging
2. post-operation - takes resulting payload on success callback as payload for logging
3. console - logs on console.
4. query - used only for query operation like data analytics